### DISCUSSION TITLE
Share same random value with multiple arbitraries

### DISCUSSION THREAD

**Discussion by coolya** [dubzzz/fast-check#3258] [Q&A] [answered]
I'm trying to test a compare function very simplified I'm trying to do something like this: 

```ts
  fc.assert(
    fc.property(
      fc.string(),
      fc.string(), 
      (a,b) => {
      const result = compare(a, b);
      return result === (a===b);
    })
  )
```

With just random values for `a` and `b` I would almost never test the case where `a` is equal to `b`.  

If could do 

```ts
  fc.assert(
    fc.property(
      fc.oneof(fc.string(), fc.constant("a")), 
      fc.oneof(fc.string(), fc.constant("a")), 
      (a,b) => {
        const result = compare(a,b);
        return result === (a===b);
      })
  )
```

But then I only test with a constant, Ideally I would like be able to describe that the two arbitraries should share the same value with a probability.

**Comment by dubzzz** [comment, marked as answer]
Hey,

Actually, with fast-check you don't really have to bother with such constraints. The case 'a === b' will not be that unlikely as the framework uses some bias internally that makes collisions more likely.

Otherwise if you really want the same value twice you may want to use `fc.clone`.

**Reply by dubzzz** [reply]
Maybe you're looking for `fc.pre(a !== b)` 🤔

**Reply by dubzzz** [reply]
@coolya Did it answer to your question or need?
